### ISSUE TITLE
Missing timeout configuration cognito client

### ISSUE THREAD

**Issue by ifero** [aws-amplify/amplify-js#7464]
**Is your feature request related to a problem? Please describe.**

I'm developing an app that does use Cognito authentication. I'm at the point in which I have to verify that the app works properly under low/no internet connection. What I did notice is that whenever I check the session with `Auth.currentSession()`, there is no internet connection AND the access token has expired, the automated API call to refresh the token hangs forever, I cannot retrieve the updated jwt token and the API call to my backend are stuck forever.

**Describe the solution you'd like**

I would like that, after a predefined amount of time (maybe passed in the configuration options), the API call fails for timeout.

**Describe alternatives you've considered**

no other options have been considered

**Additional context**
in index.js
```
Auth.configure({
   ...basicConfig,
   timeout: 5000 // or whatever time 
})
```

where_needed.jsx

```
try {
 const currentSession = Auth.currentSession()
} catch (error) {
 // could be a timeout, act as expected
}
```

**Comment by ifero** [comment]
Hi @amhinson is there any ETA on including a timeout in any Auth API call?
It is starting to become frustrating having the app hanging because there is no timeout on Auth calls

**Comment by nachozullo** [comment]
Yes, we really need a timeout in Auth requests.

**Comment by sanjamkr** [comment]
+1 – We would ideally want to have a timeout in all APIs exposed by Amplify

**Comment by tannerabread** [comment]
I just tried to reproduce this and get the call hanging indefinitely but I get different behavior. If I am in airplane/no-internet mode and try to do this, the user gets logged out automatically and I get the error `Error: Network error`

Can someone confirm that you still experience the API call hanging indefinitely? If so maybe some different reproduction steps might be helpful. For context, I was running this on a React Native project with an android emulator to test the network conditions

**Comment by stocaaro** [comment]
Related to #10393

**Comment by aliza-khu** [comment]
@tannerabread, Yes, I am also experiencing the same thing and getting Error: Network error.
